src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -400,4 +400,15 @@ object messages {
            |  ((${nestedRepresentation}))""".stripMargin
     }
   }
+
+  case class MissingReturnType()(implicit ctx:Context) extends Message(14) {
+    val kind = "Syntax"
+    val msg = "missing return type"
+    val explanation =
+      hl"""The return type is missing and needs to be clarified. For example, `area` returns a ${"Double"}:
+        |
+        |trait Shape {
+        |  def area: Double
+        |}""".stripMargin
+  }
```

Comment:
So first off, you're correctly stating that if a trait method does not have a return type it needs to be clarified. But why is this? It's an abstract definition; we need to know what it's type is when we implement it somewhere else.

I also have a different nitpick, so the error messages obey a flag called `-pagewidth X`, where `X` is the max number of columns that should be printed in the terminal. As such, I would suggest that you split line 408 to be no more than 80 characters (i.e. newline after `` `area` ``).

Good job otherwise, and thanks for cleaning up the trailing whitespace :heart: